File: src/dataset/Java/binary_exercism-java-ThomasZumsteg/claude/ClaudeSonnet4_binary_v4.java
public class Binary {
    private final String binary;
    private int cachedDecimal = -1;

    public Binary(String binary) {
        this.binary = binary;
    }

    public int getDecimal() {
        if (cachedDecimal != -1) {
            return cachedDecimal;
        }

        int result = 0;
        int length = binary.length();
        
        for (int i = 0; i < length; i++) {
            char c = binary.charAt(i);
            if (c != '0' && c != '1') {
                cachedDecimal = 0;
                return 0;
            }
            result = (result << 1) + (c - '0');
        }
        
        cachedDecimal = result;
        return result;
    }
}

File: src/dataset/c/binary_exercism-Manoloon/claude/ClaudeSonnet4_binary_v4.c
#include "binary.h"
#include <stdlib.h>

int convert(const char *input) 
{ 
    if(input == NULL) return 0;
    
    const char *ptr = input;
    size_t result = 0;
    
    while(*ptr) {
        if(*ptr == '1') {
            result = (result << 1) | 1;
        } else if(*ptr == '0') {
            result <<= 1;
        } else {
            return INVALID;
        }
        ++ptr;
    }
    
    return (int)result; 
}